packages/client/hmi-client/src/api/api.ts

```diff
@@ -28,28 +28,37 @@ API.interceptors.request.use(
 
 API.interceptors.response.use(
 	(response) => response,
-	(error) => {
-		const msg = error.response.data;
-		const status = error.response.status;
-		switch (status) {
-			case 500:
-				logger.error(msg.toString(), {
-					showToast: false,
-					toastTitle: `${ToastSummaries.SERVICE_UNAVAILABLE} (${status})`
-				});
-				break;
-			default:
-				logger.error(msg.toString(), {
-					showToast: false,
-					toastTitle: `${ToastSummaries.NETWORK_ERROR} (${status})`
-				});
-		}
-		if (status === 401) {
+	(error: AxiosError) => {
+		if (error.status === 401) {
 			// redirect to login
 			const auth = useAuthStore();
 			auth.keycloak?.login({
 				redirectUri: window.location.href
 			});
+		} else {
+			let message = error.message;
+			let title = `${error.response?.statusText} (${error.response?.status})`;
+			if (error.response && error.response.data) {
+				const responseError: ServerError = error.response?.data as ServerError;
+
+				// check to see if the 'message' property is set. If it is, set message to that value.  If not, check the 'trace' property and extract the error from that. It will be the substring between the first set of quotations marks
```

Comment:
```suggestion
				// check to see if the 'message' property is set. If it is, set message to that value.  
				// If not, check the 'trace' property and extract the error from that. 
				// It will be the substring between the first set of quotations marks.
```